src/job/jobHandler.ts

```diff
@@ -297,13 +297,20 @@ export abstract class JobHandler {
   // call this method when we want benchmarks and uses cwd option to call command outside of a one liner.
   private async wrapWithBenchmarks(
     buildStepFunc: () => Promise<CliCommandResponse>,
-    stage: string
+    stage: string,
+    logger: IJobRepoLogger
   ): Promise<CliCommandResponse> {
     const start = performance.now();
 
     const resp = await buildStepFunc();
 
     const end = performance.now();
+
+    if (resp.errorText?.length > 0) {
```

Comment:
I think this may not be a reliable check, unfortunately. 

In some instances, `errorText` will be populated even if the command doesn't actually fail. For example, the `snooty build` command writes its output to `stderr` which is what `errorText` contains. Because of this, we may throw even if the build is successful. 

I think we can either modify the commands to include the `status` property, or have them throw the error.

Reply:
Ah, I did this instead because we still need them to log an error so we know why any command failed. This at least let us not pipe through the logger, as you disliked having that throughout! So, we can't throw unless we log first. 

I think I'm confused over when the errorText would be filled. I've created multiple successful builds since this change and everything performed as expected. Can you give an example so I can test?